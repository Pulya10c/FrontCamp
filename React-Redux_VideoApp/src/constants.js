export const SM_SCREEN_SIZE = 576;
export const MD_SCREEN_SIZE = 768;

export const PATHS = {
  ANY: '*',
  HOME: '/',
  VIDEO_LIST_PAGE: '/search',
  CURRENT_VIDEO_PAGE: '/film/:id'
};

export const API = {
  SEARCH: 'search',
  SEARCH_BY: 'searchBy',
  SORT_ORDER: 'sortOrder',
  SORT_BY: 'sortBy',
  OFFSET: 'offset',
  LIMIT: 'limit'
};

export const TITLE = { text: 'Title', code: 'title' };
export const GENRES = { text: 'Genres', code: 'genres' };

export const RELEASE_DATE = { text: 'Release date', code: 'release_date' };
export const RATING = { text: 'Rating', code: 'vote_average' };

export const ASCENDING = 'asc';
export const DESCENDING = 'desc';

export const SORT_BY_VALUES = [RELEASE_DATE, RATING];
export const SEARCH_BY_VALUES = [TITLE, GENRES];
export const SORT_ORDER_VALUES = [ASCENDING, DESCENDING];
