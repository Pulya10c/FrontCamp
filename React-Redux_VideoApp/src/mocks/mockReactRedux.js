const createMockMapStateToProps = mapStateToProps => {
  if (!mapStateToProps) {
    return mapStateToProps;
  }

  return (state, ownProps) => {
    return {
      props: mapStateToProps(state, ownProps)
    };
  };
};

const getDispatchFunctionFormShorthandObject = actionsObj => dispatch => {
  return Object.keys(actionsObj).reduce((props, actionKey) => {
    return {
      ...props,
      [actionKey]: function() {
        return dispatch(actionsObj[actionKey].apply(this, arguments));
      }
    };
  }, {});
};

const createMockMapDispatchToProps = mapDispatchToProps => {
  if (!mapDispatchToProps) {
    return mapDispatchToProps;
  }

  const mappingFunction =
    typeof mapDispatchToProps === 'function'
      ? mapDispatchToProps
      : getDispatchFunctionFormShorthandObject(mapDispatchToProps);

  return (dispatch, ownProps) => {
    dispatch = dispatch || jest.fn(action => action);

    return {
      props: mappingFunction(dispatch, ownProps),
      dispatch
    };
  };
};

const createMockMergeProps = mergeProps => {
  if (!mergeProps) {
    return mergeProps;
  }

  return (stateProps, dispatchProps, ownProps) => {
    dispatchProps = dispatchProps || { dispatch: jest.fn(action => action) };

    return {
      props: mergeProps(stateProps, dispatchProps, ownProps),
      dispatch: dispatchProps.dispatch
    };
  };
};

export default {
  connect: (
    mapStateToProps,
    mapDispatchToProps,
    mergeProps
  ) => ConnectedComponent => ({
    mapStateToProps: createMockMapStateToProps(mapStateToProps),
    mapDispatchToProps: createMockMapDispatchToProps(mapDispatchToProps),
    mergeProps: createMockMergeProps(mergeProps),
    ConnectedComponent
  })
};
