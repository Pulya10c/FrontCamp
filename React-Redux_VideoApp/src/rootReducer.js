import { combineReducers } from 'redux';
import { connectRouter } from 'connected-react-router'

import videoListReducer from 'reducers/videoListReducer';
import currentVideoReducer from 'reducers/currentVideoReducer';
import uiStateReducer from 'reducers/uiStateReducer';

export const createRootReducer = (history) => combineReducers({
  router: connectRouter(history),
  uiState: uiStateReducer,
  videoList: videoListReducer,
  currentVideo: currentVideoReducer
});
