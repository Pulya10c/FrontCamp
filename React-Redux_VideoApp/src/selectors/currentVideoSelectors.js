import { createSelector } from 'reselect';
import get from 'lodash/get';

export const currentVideoSelector = state => get(state, 'currentVideo');

export const currentVideoGenresSelector = createSelector(
  [currentVideoSelector],
  currentVideo => get(currentVideo, 'genres', [])
);

export const filmPosterPathSelector = createSelector(
  [currentVideoSelector],
  currentVideo => get(currentVideo, 'poster_path') // add default img
);

export const filmTitleSelector = createSelector(
  [currentVideoSelector],
  currentVideo => get(currentVideo, 'title')
);

export const filmRatingSelector = createSelector(
  [currentVideoSelector],
  currentVideo => get(currentVideo, 'vote_average')
);

export const filmReleaseDateSelector = createSelector(
  [currentVideoSelector],
  currentVideo => get(currentVideo, 'release_date')
);

export const filmOverviewSelector = createSelector(
  [currentVideoSelector],
  currentVideo => get(currentVideo, 'overview')
);

//  currentVideo: {
//     id: 351286,
//     title: 'Jurassic World: Fallen Kingdom',
//     tagline: 'Life finds a way',
//     vote_average: 0,
//     vote_count: 28,
//     release_date: '2018-06-01',
//     poster_path: 'https://image.tmdb.org/t/p/w500/ln6d5Okr6VK5vfQVobJTiYxeD0l.jpg',
//     overview: 'A volcanic eruption threatens the remaining dinosaurs on the island of Isla Nublar, where the creatures have freely roamed for several years after the demise of an animal theme park known as Jurassic World. Claire Dearing, the former park manager, has now founded the Dinosaur Protection Group, an organization dedicated to protecting the dinosaurs. To help with her cause, Claire has recruited Owen Grady, a former dinosaur trainer who worked at the park, to prevent the extinction of the dinosaurs once again.',
//     budget: 0,
//     revenue: 0,
//     genres: [
//       'Action',
//       'Adventure',
//       'Drama',
//       'Science Fiction',
//       'Thriller'
//     ],
//     runtime: null
//   }
