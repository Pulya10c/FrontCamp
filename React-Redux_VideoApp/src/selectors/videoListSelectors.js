import { createSelector } from 'reselect'
import get from 'lodash/get';

export const videoListSelector = state => get(state, 'videoList');

export const filterConfigSelector = createSelector(
  [videoListSelector],
  videoList => get(videoList, 'filterConfig')
);

export const searchSelector = createSelector(
  [filterConfigSelector],
  filterConfig => get(filterConfig, 'search')
);

export const searchBySelector = createSelector(
  [filterConfigSelector],
  filterConfig => get(filterConfig, 'searchBy')
);
