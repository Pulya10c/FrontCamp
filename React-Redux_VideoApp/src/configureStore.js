import { createBrowserHistory } from 'history';
import { routerMiddleware } from 'connected-react-router';
import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
// import logger from 'redux-logger';

import { createRootReducer } from './rootReducer';

export const history = createBrowserHistory();

const middleware = [thunk,
// logger,
routerMiddleware(history)];
const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

export const store = createStore(
  createRootReducer(history),
  composeEnhancers(applyMiddleware(...middleware))
);
