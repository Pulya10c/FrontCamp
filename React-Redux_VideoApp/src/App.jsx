import React, { Suspense } from "react";
import PropTypes from "prop-types";
import { Redirect, Route, Switch } from "react-router-dom";
import { ConnectedRouter } from "connected-react-router";

import { history } from "src/configureStore";
import Spinner from "components/Spinner";
import MainWrapper from "components/MainWrapper";
import { PATHS } from "src/constants";

const VideoList = React.lazy(() => import("pages/VideoList"));
const CurrentVideo = React.lazy(() => import("pages/CurrentVideo"));
const NoMatch = React.lazy(() => import("pages/NoMatch"));

const App = () => (
  <ConnectedRouter history={history}>
    <MainWrapper>
      <Suspense fallback={<Spinner />}>
        <Switch>
          <Route exact path={PATHS.HOME}>
            <Redirect to={PATHS.VIDEO_LIST_PAGE} />
          </Route>
          <Route path={PATHS.VIDEO_LIST_PAGE}>
            <VideoList />
          </Route>
          <Route path={PATHS.CURRENT_VIDEO_PAGE}>
            <CurrentVideo />
          </Route>
          <Route path={PATHS.ANY}>
            <NoMatch />
          </Route>
        </Switch>
      </Suspense>
    </MainWrapper>
  </ConnectedRouter>
);

export default App;
