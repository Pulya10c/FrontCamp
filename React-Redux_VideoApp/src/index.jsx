import React from "react";
import ReactDOM from "react-dom";
import { Provider } from "react-redux";
import { ConnectedRouter } from 'connected-react-router'

import ErrorBoundary from "components/ErrorBoundary";
import App from "./App";
import { store, history } from "./configureStore";
import "./style.scss";

ReactDOM.render(
  <ErrorBoundary>
    <Provider store={store}>
      <App />
    </Provider>
  </ErrorBoundary>,
  document.getElementById("root")
);
