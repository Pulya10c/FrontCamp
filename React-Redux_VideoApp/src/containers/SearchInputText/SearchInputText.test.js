import React from 'react';
import { shallow } from 'enzyme';
import mockReactRedux from 'mocks/mockReactRedux';
import { searchSelector } from 'selectors/videoListSelectors';
import {
  updateFilterConfigBy,
  createSearchAndRedirect
} from 'actions/videoListActions';
import InputTextContainer from './';
import { API } from 'src/constants';

jest.mock('react-redux', () => mockReactRedux);

const mockSearchSelector = 'mockSearchSelector';

jest.mock('selectors/videoListSelectors', () => ({
  searchSelector: jest.fn(() => mockSearchSelector)
}));

const mockSetSearchValue = 'mockSetSearchValue';
const mockCreateSearchAndRedirect = 'mockCreateSearchAndRedirect';

jest.mock('actions/videoListActions', () => ({
  createSearchAndRedirect: jest.fn(() => mockCreateSearchAndRedirect),
  updateFilterConfigBy: {
    search: jest.fn(() => mockSetSearchValue)
  }
}));

describe('Component InputTextContainer', () => {
  const {
    mapStateToProps,
    mapDispatchToProps,
    ConnectedComponent
  } = InputTextContainer;

  it("should render correctly", () => {
    const { props } = mapStateToProps();
    expect(shallow(<ConnectedComponent {...props} />)).toMatchSnapshot();
  });

  it('should pass correct props to component from mapStateToProps', () => {
    const { props } = mapStateToProps();
    expect(props).toEqual({
      searchValue: mockSearchSelector
    });
  });

  it('should pass correct props to component from mapDispatchToProps', () => {
    const { props, dispatch } = mapDispatchToProps();
    props.createSearchAndRedirect();
    expect(createSearchAndRedirect).toHaveBeenCalled();
    expect(dispatch).toHaveBeenCalledWith(mockCreateSearchAndRedirect);

    const testValue = 'testValue';
    props.setSearchValue(testValue);
    expect(updateFilterConfigBy[API.SEARCH]).toHaveBeenCalledWith(testValue);
    expect(dispatch).toHaveBeenCalledWith(mockSetSearchValue);
  });
});
