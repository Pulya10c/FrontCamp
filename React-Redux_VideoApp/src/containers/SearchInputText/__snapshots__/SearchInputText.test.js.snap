// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component InputTextContainer should render correctly 1`] = `
<InputText
  inputValue="mockSearchSelector"
/>
`;
